javaslang/src/main/java/javaslang/collection/AbstractMap.java

```diff
@@ -21,238 +21,238 @@
  *
  * @param <K> Key type
  * @param <V> Value type
- * @param <M> Map type
+ * @param <R> Map type
  * @author Ruslan Sennov
  * @since 2.0.0
  */
-abstract class AbstractMap<K, V, M extends AbstractMap<K, V, M>> implements Map<K, V> {
+abstract class AbstractMap<K, V, R extends AbstractMap<K, V, R>> implements Map<K, V> {
```

Comment:
in my opinion they will represent the same thing.
Will rename back, if you don't agree :)